hal/src/core/usart_hal.c

```diff
@@ -39,6 +39,8 @@ typedef enum USART_Num_Def {
 #define GPIO_Remap_None 0
 
 /* Private macro -------------------------------------------------------------*/
+// IS_USART_CONFIG_VALID(config) - returns true for 8 data bit, any flow control, any parity, any stop byte configurations
+#define IS_USART_CONFIG_VALID(CONFIG) ( (((CONFIG & 0b00001100)>>2) != 0b11) && (((CONFIG & 0b00110000)>>4)==0b11) )
```

Comment:
I second the question. `(CONFIG & 0b00110000)>>4)==0b11` part doesn't seem to check anything that is used in `HAL_USART_BeginConfig()` later, and fails for `SERIAL_8N1`.